src/app/hooks/usePan.ts

```diff
@@ -31,32 +32,42 @@ export const usePan = (active: boolean) => {
 
       return { translateX: mX, translateY: mY };
     });
-  };
+  }, []);
 
-  const handleMouseUp = (evt: MouseEvent) => {
-    evt.preventDefault();
-    setCursor('grab');
+  const handleMouseUp = useCallback(
+    (evt: MouseEvent) => {
+      evt.preventDefault();
+      setCursor('grab');
+      setStatus('idle');
 
-    document.removeEventListener('mousemove', handleMouseMove);
-    document.removeEventListener('mouseup', handleMouseUp);
-  };
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    },
+    [handleMouseMove]
+  );
 
-  const handleMouseDown: MouseEventHandler<HTMLElement> = (evt) => {
-    if (!active) return;
-    evt.preventDefault();
-    setCursor('grabbing');
+  const handleMouseDown = useCallback(
+    (evt: React.MouseEvent<HTMLElement>) => {
+      if (!active) return;
+      evt.preventDefault();
+      setCursor('grabbing');
+      setStatus('moving');
 
-    document.addEventListener('mousemove', handleMouseMove);
-    document.addEventListener('mouseup', handleMouseUp);
-  };
+      document.addEventListener('mousemove', handleMouseMove);
+      document.addEventListener('mouseup', handleMouseUp);
+    },
+    [active, handleMouseMove, handleMouseUp]
+  );
 
   useEffect(() => {
     if (!active) setPan(INITIAL_PAN);
   }, [active]);
 
   return {
     pan,
+    setPan,
```

Comment:
I think it will be a good idea to move pan useState outside hook, just like you did with `useWheelZoom` hook and have onChange callback.